extensions/amp-subscriptions/0.1/entitlement-store.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {Observable} from '../../../src/observable';
+
+/** @typedef {{serviceId: string, entitlements: ./entitlements.Entitlements}} */
+export let EntitlementChangeEventDef;
+
+export class EntitlementStore {
+  /**
+   *
+   * @param {!Array<string>} expectedServiceIds
+   */
+  constructor(expectedServiceIds) {
+
+    /** @private @const {!Array<string>} */
+    this.serviceIds_ = expectedServiceIds;
+
+    /** @private @const Object<serviceId: string, entitlements: ./entitlements.Entitlements> */
+    this.entitlements_ = {};
+
+    /** @private @const {Observable<!EntitlementChangeEventDef>} */
+    this.onChangeCallbacks_ = new Observable();
+
+    /** @private {?Promise<./entitlements.Entitlements>} */
```

Comment:
Must have `!./ent...` - I.e. promise always yield some response. Otherwise this implies nullable.